Index: project/libs/build/compilated/_akimage.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- project/libs/build/compilated/_akimage.js	(date 1382660925000)
+++ project/libs/build/compilated/_akimage.js	(revision )
@@ -253,9 +253,11 @@
 		};
 
 
+/*
+* AKHistogram
+* */
 
 
-
     Akimage.AkHistogram = function(){
 
         this.AkHistogram = {
@@ -655,6 +657,78 @@
                 };
 
 
+
+
+
+                /**	 * @function {AkCreateHist} Create a ROI rectangle
+                 *
+                 **/
+                /**
+                 * 		@param {number} _xOffset X offset
+                 * 	@param {number} _yOffset X offset
+                 * 	@param {number} _Width X offset
+                 * 	@param {number} _Height X offset
+                 *      @return {AIROI}
+                 * */
+
+
+                _Akontext.AkCreateHist = function(_bins) {
+
+                    //if (arguments.length!=4) {AKerrors[5]= true; AKLastError=5; throw "invalid number of arguments";return false;}
+
+                    //if (_xOffset < 0 || _yOffset <0|| _Width <1|| _Height<1) {AKerrors[4]= true; AKLastError=4; throw "invalid value of argument" ;return false;}
+
+
+                   var multi = false;
+                   if(_bins[0][0] != undefined) {multi = true;}
+
+
+
+
+                    var _Histogram = (new Akimage.AkHistogram()).AkHistogram;
+
+
+                    if(!multi){
+
+                        if(_bins[1]<_bins[0]){
+                            return "error";
+                        }
+
+                        for(var k = _bins[0]; k<_bins[1];k++){
+
+
+                            _Histogram.bins[k] = k;
+                        }
+
+                    }
+
+
+
+                    if(multi){
+
+                        var _i = 0;
+
+                        for(var p= 0; p<_bins.length;p++){
+
+                            if(_bins[p][1]<_bins[p][0]){
+                                return "error";}
+
+                            for(var k = _bins[p][0]; k<_bins[p][1];k++){
+
+
+                                _Histogram.bins[_i] = p;
+                                _i++;
+                            }
+
+                        }
+
+                    }
+
+
+                    return (_Histogram);
+
+                };
+
 				})(this);
 
 
@@ -708,13 +782,13 @@
 
             (function (_Akontext) {
 
-            /**	 * @function AkCreateHist set a ROI in the Akimage passed by arguments
+            /**	 * @function AkCalcHist set a ROI in the Akimage passed by arguments
 				 * @param {Akimage} ImageReference Object
                 * @param  {AROI} Region of interest
 			**/
 
 
-					_Akontext.AkCreateHist = function(_ImIn, _ROI) {
+					_Akontext.AkCalcHist = function(_ImIn, _ROI) {
 
                         _ImIn.roi = _ROI;
 
Index: project/libs/build/03_constructors.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- project/libs/build/03_constructors.js	(date 1382660925000)
+++ project/libs/build/03_constructors.js	(revision )
@@ -392,13 +392,11 @@
 
                 _Akontext.AkCreateHist = function(_bins) {
 
-                    //if (arguments.length!=4) {AKerrors[5]= true; AKLastError=5; throw "invalid number of arguments";return false;}
+                    if (arguments.length!=1) {AKerrors[5]= true; AKLastError=5; throw "invalid number of arguments";return false;}
+                    if (!(Object.prototype.toString.apply(_bins) === '[object Array]')) {AKerrors[19]= true; throw "In Histogram array expeted" AKLastError=19; return false;}
 
-                    //if (_xOffset < 0 || _yOffset <0|| _Width <1|| _Height<1) {AKerrors[4]= true; AKLastError=4; throw "invalid value of argument" ;return false;}
-
-
                    var multi = false;
-                   if(bins[0][0] != undefined) multi = true;
+                   if(_bins[0][0] != undefined) {multi = true;}
 
 
 
@@ -408,10 +406,10 @@
 
                     if(!multi){
 
-                        if(_bins[1]<_bins[0]){
-                            return "error";
-                        }
+                        if(_bins[1]<_bins[0])
+                            {AKerrors[20]= true; AKLastError=20; throw "in Histogram invalid hight value is low than low value";return false;}
 
+
                         for(var k = _bins[0]; k<_bins[1];k++){
 
 
@@ -428,22 +426,19 @@
 
                         for(var p= 0; p<_bins.length;p++){
 
-                            if(_bins[p][1]<_bins[p][0]){
-                                return "error";}
+                            if(_bins[p][1]<_bins[p][0])
+                                {AKerrors[20]= true; AKLastError=20; throw "in Histogram invalid hight value is low than low value";return false;}
 
                             for(var k = _bins[p][0]; k<_bins[p][1];k++){
 
 
                                 _Histogram.bins[_i] = p;
+                                _i++;
                             }
 
                         }
 
                     }
-
-
-
-
 
 
                     return (_Histogram);
